File: src/net/teamfps/java/serialization/Type.java
/**
 * 
 */
package net.teamfps.java.serialization;

/**
 * @author Zekye
 *
 */
public enum Type {
	UNKNOWN(0, 0), BYTE(1, 1), SHORT(2, 2), CHAR(3, 2), INT(4, 4), LONG(5, 8), FLOAT(6, 4), DOUBLE(7, 8), BOOLEAN(8, 1), STRING(9, 1);
	private final byte type;

	private final int size;

	private Type(int type, int size) {
		this.type = (byte) type;
		this.size = size;
	}

	/**
	 * @return the id
	 */
	public byte getType() {
		return type;
	}

	/**
	 * @return the size
	 */
	public int getSize() {
		return size;
	}

	public static Type getType(byte type) {
		for (Type t : values()) {
			if (t.getType() == type) {
				return t;
			}
		}
		assert (false);
		return UNKNOWN;
	}

}

File: src/net/teamfps/cplus/serialization/Type.cpp
#include "Type.h"
/**
* @author Zekye
*
*/
namespace teamfps {
	const Type Type::UNKNOWN = Type(0, 0);
	const Type Type::BYTE = Type(1, 1);
	const Type Type::SHORT = Type(2, 2);
	const Type Type::CHAR = Type(3, 1);
	const Type Type::INT = Type(4, 4);
	const Type Type::LONG = Type(5, 8);
	const Type Type::FLOAT = Type(6, 4);
	const Type Type::DOUBLE = Type(7, 8);
	const Type Type::BOOLEAN = Type(8, 1);
	const Type Type::STRING = Type(9, 1);

	Type::Type(ubyte type, uint size) : type(type), size(size) {}

	const ubyte Type::getType() const {
		return type;
	}

	const uint Type::getSize() const {
		return size;
	}
}